src/lightning/pytorch/trainer/configuration_validator.py

```diff
@@ -152,16 +153,19 @@ def __verify_manual_optimization_support(trainer: "pl.Trainer", model: "pl.Light
         )
 
 
-def __check_training_step_requires_dataloader_iter(model: "pl.LightningModule") -> None:
-    """Check if the current `training_step` is requesting `dataloader_iter`."""
-    if is_param_in_hook_signature(model.training_step, "dataloader_iter", explicit=True):
-        for hook in ("on_train_batch_start", "on_train_batch_end"):
-            if is_overridden(hook, model):
-                rank_zero_warn(
-                    f"The `batch_idx` argument in `{type(model).__name__}.{hook}` hook may"
-                    " not match with the actual batch index when using a `dataloader_iter`"
-                    " argument in your `training_step`."
-                )
+def __warn_dataloader_iter_limitations(model: "pl.LightningModule") -> None:
```

Comment:
This change is actually unrelated to the bugfix, but the bugfix made me think that `dataloader_iter` will also be broken with gradient accumulation.

Unfortunately, there's no way for us to know whether the user will fetch more than one batch per step. Given how tricky and experimental this feature is, I would say it's good to be this vocal about the limitations.